### PR TITLE
[ASV-1757] Helping wallet install metrics

### PR DESCRIPTION
**Pull request by phillipsgoncalves**
**What does this PR do?**

   This PR aims at implementing the metrics needed to the helping AppCoins wallet installation feature.
Also fixed the context used on the download complete event that is sent facebook. Previously we were using the previousContext instead of the currentContext meaning that we are going to send the previous view instead of the current view where the download is starting. 

Also fixed the Install and Download events sent to the BI by adding the appc flags, which were not added in ASV-1591.

**Database changed?**

No

**Where should the reviewer start?**

- [ ] WalletInstallAnalytics.java

**How should this be manually tested?**

  Perform wallet download and installation and see the events being sent in both Charles (BI) and analytics manager logs.

**What are the relevant tickets?**

  Tickets related to this pull-request: [ASV-1757](https://aptoide.atlassian.net/browse/ASV-1757)

**Questions:**

   Does this add new dependencies which need to be added? (Eg. new keys, etc.) 

**Code Review Checklist**

- [ ] Documentation on public interfaces
- [ ] Database changed? If yes - Migration?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] New Kotlin code has unit tests
- [ ] New flows in presenters unit tests
- [ ] Mappers/Validators with any kind of logic unit tests
- [ ] Functional tests pass

### CODE REVIEW

File: app/src/main/java/cm/aptoide/pt/wallet/WalletInstallAnalytics.kt [hunk 1]
```diff
@@ -39,9 +52,37 @@ class WalletInstallAnalytics(val downloadAnalytics: DownloadAnalytics,
...
+    if (shouldRegister) {
+      if (historyTracker == null) {
+        throw RuntimeException("If "
+            + this.javaClass
+            .simpleName
+            + " should be logged to screen history, it has to return a value on method NavigationTrackFragment#getHistoryTracker")
+      }
+      navigationTracker.registerScreen(historyTracker)
+    }
+    shouldRegister = false;
```
Comment (franciscoaleixo) [inline app/src/main/java/cm/aptoide/pt/wallet/WalletInstallAnalytics.kt]: ";"

File: app/src/main/java/cm/aptoide/pt/wallet/WalletInstallAnalytics.kt [hunk 2]
```diff
@@ -39,9 +52,37 @@ class WalletInstallAnalytics(val downloadAnalytics: DownloadAnalytics,
...
+  fun sendClickOnInstallButtonEvent(packageName: String, applicationPublisher: String) {
+    val map = HashMap<String, Any>()
+    map[TYPE] = "Install"
+    map[APPLICATION_NAME] = packageName
+    map[APPLICATION_PUBLISHER] = applicationPublisher
+    analyticsManager.logEvent(map, CLICK_INSTALL, AnalyticsManager.Action.CLICK,
+        VIEW_CONTEXT)
+  }
+
+  fun setupHistoryTracker() {
```
Comment (franciscoaleixo) [inline app/src/main/java/cm/aptoide/pt/wallet/WalletInstallAnalytics.kt]: Why is this here? If we want to register the screen, shouldn't it be in the activity?

Comment (phillipsgoncalves) [reply app/src/main/java/cm/aptoide/pt/wallet/WalletInstallAnalytics.kt]: This was done by suggestion from João.
Ideally we would have a base activity that would do this management, just like we have the NavigationTrackerFragment, but João said it was not worth the effort, considering we only have one more activity which is the MainActivity. That is why every other navigation track management is done in the respective Fragments.
For this case we did it here in the wallet analytics because we needed to add a flag (shouldRegister) and we had to add the dependency of NavigationTracker. It didn't make sense to make the activity dirty because of this.

Comment (franciscoaleixo) [reply app/src/main/java/cm/aptoide/pt/wallet/WalletInstallAnalytics.kt]: 👍
